vkcs/internal/services/networking/v2/routers/requests.go

```diff
@@ -18,10 +24,61 @@ func (opts RouterCreateOpts) ToRouterCreateMap() (map[string]interface{}, error)
 	return util.BuildRequest(opts, "router")
 }
 
-func Create(c *gophercloud.ServiceClient, opts routers.CreateOptsBuilder) routers.CreateResult {
-	r := routers.Create(c, opts)
-	r.Err = util.ErrorWithRequestID(r.Err, r.Header.Get(util.RequestIDHeader))
-	return r
+const creationRetriesNum = 3
```

Comment:
I'd like to see a generic utility function similar to `retry.RetryContext` in `terraform-provider-sdkv2` but with options for our needs, specifically, number of retry attempts and delay between attempts. We could use such a function with closures inside `Create` functions for resources specifying that error is retryable, for example, when response has status code 409 (for this check you could also use our [utility](https://github.com/vk-cs/terraform-provider-vkcs/blob/master/vkcs/internal/util/errutil/err.go)) and some specific neutron error in body. You could name this "check" function `retryNeutronError`, so it could accept status codes and neutron error types and return whether error is retryable or not. This way it can be used in a similar way across all networking resources where it's needed.

Reply:
done